### PR TITLE
Extract RecordWorkflowTaskStartedRequest handler

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Extracted handler for `RecordWorkflowTaskStarted` out of `workflow_task_handler_callbacks.go` into its own file.

## Why?

There's no clear benefit to grouping the WFT API handlers all in the - poorly named - file. The PR makes this consistent with the existing API handlers now.

## How did you test it?

Pure refactoring, no behavior change.

### CODE REVIEW

File: service/history/api/verifychildworkflowcompletionrecorded/api.go [hunk 1]
```diff
@@ -34,14 +34,12 @@ import (
...
 	"go.temporal.io/server/service/history/api"
 	"go.temporal.io/server/service/history/consts"
-	"go.temporal.io/server/service/history/shard"
 	"go.temporal.io/server/service/history/workflow"
 )
 
 func Invoke(
 	ctx context.Context,
 	request *historyservice.VerifyChildExecutionCompletionRecordedRequest,
-	shard shard.Context,
```
Comment (stephanos) [inline service/history/api/verifychildworkflowcompletionrecorded/api.go:44]: drive-by fix: this wasn't actually used

File: service/history/workflow_task_handler_callbacks.go [hunk 2]
```diff
@@ -765,7 +620,14 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 	resp := &historyservice.RespondWorkflowTaskCompletedResponse{}
 	//nolint:staticcheck
 	if request.GetReturnNewWorkflowTask() && newWorkflowTask != nil {
-		resp.StartedResponse, err = handler.createRecordWorkflowTaskStartedResponse(ctx, ms, updateRegistry, newWorkflowTask, request.GetIdentity(), request.GetForceCreateNewWorkflowTask())
+		resp.StartedResponse, err = recordworkflowtaskstarted.CreateRecordWorkflowTaskStartedResponse(
```
Comment (stephanos) [inline service/history/workflow_task_handler_callbacks.go:623]: This now lives in `recordworkflowtaskstarted`

File: service/history/api/recordworkflowtaskstarted/api.go
Comment (stephanos) [inline service/history/api/recordworkflowtaskstarted/api.go:1]: Code as exactly as before, except references to `handler` had to be replaced, obviously.

**Review by dandavison — APPROVED** [review]
Beautiful. This all seems to me like it will be a big improvement.
